src/__tests__/data/SubComponent.tsx

```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+
+export default function Root(props: { name: string }) {
+  return <span>root {props.name}</span>;
+}
+Root.displayName = 'Root';
```

Comment:
I had to explicitly set `displayName` here because, if I don't, the docgen incorrectly names this component `SubComponent` (naming it after the file). I feel like it's tangentially related to https://github.com/styleguidist/react-docgen-typescript/issues/395.